User Story 1981979

TomArcherMsft opened the pull request:
[User Story 1981979](https://dev.azure.com/mseng/TechnicalContent/_workitems/edit/1981979)

TomArcherMsft committed:
Fixing article

TomArcherMsft committed:
Updating main to use new rg var name

TomArcherMsft committed:
Updating main to use new rg var name

TomArcherMsft committed:
Removed deprecated block and fixed sku value

TomArcherMsft committed:
Marked various output values as sensitive

TomArcherMsft committed:
readme files

TomArcherMsft committed:
readme files

lonegunmanb commented on quickstart/201-k8s-cluster-with-tf-and-aks/outputs.tf:
```diff
@@ -0,0 +1,39 @@
+}
+
+output "client_key" {
+  value     = azurerm_kubernetes_cluster.k8s.kube_config.0.client_key
```
Can we change the dot index syntax to the new square bracket syntax due to the [community rule](https://github.com/terraform-linters/tflint/blob/v0.39.3/docs/rules/terraform_deprecated_index.md)?

```hcl
value     = azurerm_kubernetes_cluster.k8s.kube_config[0].client_key
```

lonegunmanb commented on quickstart/201-k8s-cluster-with-tf-and-aks/outputs.tf:
```diff
@@ -0,0 +1,39 @@
+}
+
+output "client_certificate" {
+  value     = azurerm_kubernetes_cluster.k8s.kube_config.0.client_certificate
```
Can we change the dot index syntax to the new square bracket syntax due to the [community rule](https://github.com/terraform-linters/tflint/blob/v0.39.3/docs/rules/terraform_deprecated_index.md)?

```hcl
value     = azurerm_kubernetes_cluster.k8s.kube_config[0].client_certificate
```

lonegunmanb commented on quickstart/201-k8s-cluster-with-tf-and-aks/outputs.tf:
```diff
@@ -0,0 +1,39 @@
+}
+
+output "cluster_ca_certificate" {
+  value     = azurerm_kubernetes_cluster.k8s.kube_config.0.cluster_ca_certificate
```
Can we change the dot index syntax to the new square bracket syntax due to the [community rule](https://github.com/terraform-linters/tflint/blob/v0.39.3/docs/rules/terraform_deprecated_index.md)?

```hcl
value     = azurerm_kubernetes_cluster.k8s.kube_config[0].cluster_ca_certificate
```

lonegunmanb commented on quickstart/201-k8s-cluster-with-tf-and-aks/outputs.tf:
```diff
@@ -0,0 +1,39 @@
+}
+
+output "cluster_username" {
+  value     = azurerm_kubernetes_cluster.k8s.kube_config.0.username
```
Can we change the dot index syntax to the new square bracket syntax due to the [community rule](https://github.com/terraform-linters/tflint/blob/v0.39.3/docs/rules/terraform_deprecated_index.md)?

```hcl
value     = azurerm_kubernetes_cluster.k8s.kube_config[0].username
```

lonegunmanb commented on quickstart/201-k8s-cluster-with-tf-and-aks/outputs.tf:
```diff
@@ -0,0 +1,39 @@
+}
+
+output "cluster_password" {
+  value     = azurerm_kubernetes_cluster.k8s.kube_config.0.password
```
Can we change the dot index syntax to the new square bracket syntax due to the [community rule](https://github.com/terraform-linters/tflint/blob/v0.39.3/docs/rules/terraform_deprecated_index.md)?

```hcl
value     = azurerm_kubernetes_cluster.k8s.kube_config[0].password
```

lonegunmanb commented on quickstart/201-k8s-cluster-with-tf-and-aks/outputs.tf:
```diff
@@ -0,0 +1,39 @@
+}
+
+output "host" {
+  value     = azurerm_kubernetes_cluster.k8s.kube_config.0.host
```
Can we change the dot index syntax to the new square bracket syntax due to the [community rule](https://github.com/terraform-linters/tflint/blob/v0.39.3/docs/rules/terraform_deprecated_index.md)?

```hcl
value     = azurerm_kubernetes_cluster.k8s.kube_config[0].host
```

lonegunmanb commented on quickstart/201-k8s-cluster-with-tf-and-aks/providers.tf:
```diff
@@ -0,0 +1,14 @@
+      source  = "hashicorp/azurerm"
+      version = "~>3.0"
+    }
+  }
```
Can we add `random` provider version here? Such as:

```hcl
random = {
      source = "hashicorp/random"
      version = "~>3.0"
}
```

lonegunmanb commented on quickstart/201-k8s-cluster-with-tf-and-aks/variables.tf:
```diff
@@ -0,0 +1,53 @@
+  default = ""
+}
+
+variable "aks_service_principal_object_id" {
```
Since this variable is only referenced in `terraform.tfvars`, should we just remove it?

lonegunmanb commented on quickstart/201-mysql-fs-db/main.tf:
```diff
@@ -20,21 +25,15 @@ resource "random_password" "password" {
+# Manages the Virtual Network
 resource "azurerm_virtual_network" "default" {
   name                = "vnet-${random_string.name.result}"
   location            = azurerm_resource_group.default.location
```
Since the `azurerm_resource_group.default` has been removed, I guess we should change this line to:

```hcl
location            = azurerm_resource_group.rg.location
```

lonegunmanb commented on quickstart/201-mysql-fs-db/main.tf:
```diff
@@ -55,21 +54,21 @@ resource "azurerm_subnet" "default" {
+# Enables you to manage Private DNS zones within Azure DNS
 resource "azurerm_private_dns_zone" "default" {
   name                = "${random_string.name.result}.mysql.database.azure.com"
   resource_group_name = azurerm_resource_group.default.name
```
Since the `azurerm_resource_group.default` has been removed, I guess we should change this line to:

```hcl
resource_group_name = azurerm_resource_group.rg.name
```

lonegunmanb commented on quickstart/201-mysql-fs-db/main.tf:
```diff
@@ -55,21 +54,21 @@ resource "azurerm_subnet" "default" {
   name                  = "mysqlfsVnetZone${random_string.name.result}.com"
   private_dns_zone_name = azurerm_private_dns_zone.default.name
   virtual_network_id    = azurerm_virtual_network.default.id
   resource_group_name   = azurerm_resource_group.default.name
```
Since the `azurerm_resource_group.default` has been removed, I guess we should change this line to:

```hcl
resource_group_name = azurerm_resource_group.rg.name
```

lonegunmanb commented on quickstart/201-mysql-fs-db/main.tf:
```diff
@@ -55,21 +54,21 @@ resource "azurerm_subnet" "default" {
+# Manages the MySQL Flexible Server
 resource "azurerm_mysql_flexible_server" "default" {
   name                         = "mysqlfs-${random_string.name.result}"
   resource_group_name          = azurerm_resource_group.default.name
```
Since the `azurerm_resource_group.default` has been removed, I guess we should change this line to:

```hcl
resource_group_name = azurerm_resource_group.rg.name
```

lonegunmanb replied:
Same to the next line:

```hcl
location                     = azurerm_resource_group.rg.location
```

lonegunmanb commented on quickstart/201-mysql-fs-db/main.tf:
```diff
@@ -1,17 +1,22 @@
 }
 
-// Generate random value for the name
+resource "azurerm_resource_group" "rg" {
```
We also need change the resource group name we used in `mysql-fs-db.tf` file:

```hcl
// Manages the MySQL Flexible Server Database
resource "azurerm_mysql_flexible_database" "default" {
  name                = "mysqlfsdb_${random_string.name.result}"
  resource_group_name = azurerm_resource_group.rg.name
  server_name         = azurerm_mysql_flexible_server.rg.name # We should rename default to rg here
  charset             = "utf8"
  collation           = "utf8_unicode_ci"
}
```

lonegunmanb replied:
Same in `outputs.tf` file:

```hcl
output "resource_group_name" {
  value = azurerm_resource_group.rg.name # We should rename default to rg here
}
```

lonegunmanb commented on quickstart/201-mysql-fs-db/main.tf:
```diff
@@ -1,17 +1,22 @@
-// Generate random value for the Resource Group name
+# Generate random resource group name
 resource "random_pet" "rg_name" {
```
Since we've used `random` resource here, could we add the following block in `providers.tf` file?

```hcl
random = {
      source = "hashicorp/random"
      version = "~>3.0"
}
```

lonegunmanb commented on quickstart/201-mysql-fs-db/main.tf:
```diff
@@ -1,17 +1,22 @@
+  location = var.resource_group_location
+}
+
+# Generate random value for the name
```
Since we're using `#` as single line comment for this module, could we change line 10 in `mysql-fs-db.tf` file?

```hcl
 // Manages the MySQL Flexible Server Database
```

to

```hcl
 # Manages the MySQL Flexible Server Database
```

lonegunmanb commented on quickstart/201-mysql-fs-db/main.tf:
```diff
@@ -1,17 +1,22 @@
   length  = 8
   upper   = false
   lower   = true
   special = false
```
We might add the following line there, otherwise the generated name might start with a number, that will be an invalid admin user name for `azurerm_mysql_flexible_server.default` resource.

```hcl
  numeric  = false
```

lonegunmanb reviewed (changes requested):
Hi @TomArcherMsft , we have some issues here, would you please take a look? Thanks！

lonegunmanb commented on quickstart/201-mysql-fs-db/main.tf:
```diff
@@ -20,21 +25,15 @@ resource "random_password" "password" {
 resource "azurerm_virtual_network" "default" {
   name                = "vnet-${random_string.name.result}"
   location            = azurerm_resource_group.default.location
   resource_group_name = azurerm_resource_group.default.name
```
Since the `azurerm_resource_group.default` has been removed, I guess we should change this line to:

```hcl
resource_group_name           = azurerm_resource_group.rg.name
```

lonegunmanb commented on quickstart/201-mysql-fs-db/main.tf:
```diff
@@ -20,21 +25,15 @@ resource "random_password" "password" {
+# Manages the Subnet
 resource "azurerm_subnet" "default" {
   name                 = "subnet-${random_string.name.result}"
   resource_group_name  = azurerm_resource_group.default.name
```
Since the `azurerm_resource_group.default` has been removed, I guess we should change this line to:

```hcl
resource_group_name  = azurerm_resource_group.rg.name
```

TomArcherMsft committed:
changes per tech review

TomArcherMsft committed:
fixing random provider info

TomArcherMsft committed:
changing array syntax

TomArcherMsft committed:
more changes

TomArcherMsft committed:
more changes

lonegunmanb committed:
Sort all variables, outputs, arguments and blocks in resource block.
Adjust code file's style.
Add `numeric = false` for `random_string.name` in `201-mysql-fs-db` module as the name is used as `azurerm_mysql_flexible_server.default`'s `administrator_login` and the login name doesn't support a name start with number.
Add minimum constraints for `random_password.password` in `201-mysql-fs-db` module as the password required a minimum complexity.

TomArcherMsft committed:
Merge pull request #1 from lonegunmanb/UserStory1981979_zj

Re-org code structure and fix a potential bug caused by `random_password.password`.

TomArcherMsft commented:
#sign-off

lonegunmanb reviewed (approved):
Thanks @TomArcherMsft , LGTM!
